src/components/Q&A/qList.jsx

```diff
@@ -0,0 +1,279 @@
+      show: false,
+      currentID: '',
+      sort: false
+    }
+
+    this.sortQuestions = this.sortQuestions.bind(this);
+    this.handleSearch = this.handleSearch.bind(this);
+    this.handleSeeLessQues = this.handleSeeLessQues.bind(this);
+    this.handleSeeMoreQues = this.handleSeeMoreQues.bind(this);
+    this.handleAddQues = this.handleAddQues.bind(this);
+    this.handleClose = this.handleClose.bind(this);
+  }
+
+  sortQuestions() {
+      var qArr = this.state.productQ.results;
+      if (qArr.length > 1) {
+        // for (var i = 0; i < qArr.length; i++) {
+        //   if(qArr[i-1]) {
+        //     if(qArr[i].question_helpfulness > qArr[0].question_helpfulness) {
+        //       var temp = qArr[i];
+        //       qArr[i] = qArr[i-1];
+        //       qArr[i-1] = temp;
+        //     }
+        //   }
+        // }
+        qArr.sort((a, b) => {
+          return b.question_helpfulness - a.question_helpfulness;
+        });
+        this.setState({totalQuesArr: qArr});
+        var temp = qArr.slice(0,2);
+        this.setState({questionArr: temp});
+      }
+      this.setState({totalQuesArr: qArr});
+      var tempArr = qArr.slice(0,2);
+      this.setState({questionArr: tempArr});
+  }
+
+  handleSearch(e) {
+    var term = e.target.value;
+    term = term.toLowerCase();
+    var searchArr = this.state.totalQuesArr;
+    if (term.length >= 3) {
+      this.setState({searchSort: true});
+      for(var i = 0; i < searchArr.length; i++) {
+        if (searchArr[i - 1]) {
+          var currentStr = searchArr[i].question_body;
+          if (currentStr.includes(term)) {
+            var temp = searchArr[i];
+            searchArr[i] = searchArr[0];
+            searchArr[0] = temp;
+          }
+        }
+      }
+      this.setState({totalQuesArr: searchArr});
+      var temp = searchArr.slice(0,2); //make changes here when using live data
+      this.setState({searchArray: temp});
+    } else {
+      this.setState({searchSort: false});
+      this.sortQuestions();
+    }
+  }
+
+  handleSeeMoreQues(totalArr) {
+    if(this.state.searchSort) {
+      this.setState({btnClick: true}); //on each button click should add 2 more questions to arr
+      this.setState({searchArray: totalArr}); //TODO: once working with live data make changes here
+    } else {
+      this.setState({btnClick: true});
+      this.setState({questionArr: totalArr});
+    }
+  }
+
+  handleSeeLessQues(totalArr) {
+    if(this.state.searchSort) {
+      this.setState({btnClick: false});
+      var temp = totalArr.slice(0, 2);
+      this.setState({searchArray: temp});
+    } else {
+      this.setState({btnClick: false});
+      var tempArr = totalArr.slice(0, 2);
+      this.setState({questionArr: tempArr});
+    }
+  }
+
+  handleAddQues() {
+    this.setState({addClick: true});
+  }
+
+  handleClose(e) {
+    e.preventDefault();
+    this.setState({show: false})
+  }
+//use componentDidUpdate with a conditional statement
+  componentDidUpdate() {
```

Comment:
For mine, I needed to pass 'prevProps' as an argument. Otherwise it was re-rendering on every single state/prop change. If you pass in prevProps, it takes a snapshot  of current props/state and will only re-render if something you specified changes. Not sure if/how that applies here, but just a thought.

Reply:
Great tip! I ended up using it and it works much better with my component.